pilot/pkg/security/model/authentication.go

```diff
@@ -285,6 +287,33 @@ func ApplyToCommonTLSContext(tlsContext *tls.CommonTlsContext, proxy *model.Prox
 	}
 }
 
+// SANToExactMatchWithFallback creates prefix matchs as fallbacks in addition to the converted exact matches,
+// if SAN inputs are SPIFFE URIs and end with the valid wildcard.
+func SANToExactMatchWithFallback(in []string) []*matcher.StringMatcher {
+	if len(in) == 0 {
+		return nil
+	}
+
+	res := make([]*matcher.StringMatcher, 0, len(in))
+	for _, s := range in {
+		// Replace e.g. "foo" with a exact match
+		res = append(res, &matcher.StringMatcher{
+			MatchPattern: &matcher.StringMatcher_Exact{Exact: s},
```

Comment:
do we want exact match AND prefix match? I would have thought its one or the other

Reply:
Agree, I think one or the other would suffice.

Reply:
Updated, thanks!